### PR TITLE
chore: log daemon stdout and stderr

### PR DESCRIPTION
**Pull request by achingbrain**
I find myself making this change all the time to see logs from the daemon when writing interop tests.

With this change to see output from the daemon you can do:

```sh
$ DEBUG=ipfsd-ctl:daemon:* npm test
```

### CODE REVIEW

File: src/ipfsd-daemon.js [hunk 1]
```diff
@@ -246,13 +250,21 @@ class Daemon {
...
+        const input = line
           .split('\n')
           .map((l) => l.trim())
           .filter(Boolean)
           .slice(-1)[0] || ''
 
+        line = line.trim()
+
+        if (line) {
+          daemonLog.err(line)
```
Comment (vasco-santos) [inline src/ipfsd-daemon.js]: Do you have a particular reason for doing the string trim before? 

I think that the only way it should be used, is if we can receive error strings composed by empty spaces. If this is not the case, what about having the trim only in the log line?

File: src/ipfsd-daemon.js [hunk 2]
```diff
@@ -246,13 +250,21 @@ class Daemon {
     let output = ''
     this.subprocess = run(this, args, { env: this.env }, {
       error: (err) => {
+        let line = String(err)
```
Comment (vasco-santos) [inline src/ipfsd-daemon.js]: The name `line` seems kind of unrelated. What do you think of `errString`, or something more related to the string content?

File: src/ipfsd-daemon.js [hunk 3]
```diff
@@ -263,7 +275,15 @@ class Daemon {
         }
       },
       data: (data) => {
-        output += String(data)
+        let line = String(data)
```
Comment (vasco-santos) [inline src/ipfsd-daemon.js]: The name `line` seems kind of unrelated. What do you think of `dataString`, or something more related to the string content?

File: src/ipfsd-daemon.js (hunk 1 above)
Comment (achingbrain) [reply src/ipfsd-daemon.js]: The err string has a newline at the end and `debug` will add a newline to anything it's passed so the trim just removes it.  The var isn't used after this so it was just to not do:

```javascript
if (line.trim()) {
   daemonLog.err(line.trim())
}
```

If you prefer I could just do:

```javascript
if (line) {
   daemonLog.err(line.trim())
}
```

File: src/ipfsd-daemon.js (hunk 2 above)
Comment (achingbrain) [reply src/ipfsd-daemon.js]: `err` is a string with a newline character at the end, hence `line`.

`errString` to me would be better as `errorString` but then I've never been a fan of Hungarian notation so I don't think we should encode the type in variable names and certainly not in a language without types, so just `error`.

...But then by convention we'd expect `error` to be an `Error` object, so it confuses the purpose of this callback.  I'd prefer the options to be:

```javascript
{
  onStdErr: (line) => {},
  onStdOut: (line) => {}
}
```

instead of:

```javascript
{
  error: (err) => {},
  data: (data) => {}
}
```

File: src/ipfsd-daemon.js (hunk 3 above)
Comment (achingbrain) [reply src/ipfsd-daemon.js]: See previous comment.

File: src/ipfsd-daemon.js (hunk 1 above)
Comment (vasco-santos) [reply src/ipfsd-daemon.js]: yes, I would go for your second code snippet.

File: src/ipfsd-daemon.js (hunk 2 above)
Comment (vasco-santos) [reply src/ipfsd-daemon.js]: I see your point. I would suggest changing it to `errorLine` or keep as you have it.

File: src/ipfsd-daemon.js (hunk 1 above)
Comment (achingbrain) [reply src/ipfsd-daemon.js]: Done.
